### PR TITLE
[SpawnageLoong] iP

### PR DESCRIPTION
**Pull request by SpawnageLoong**

### CODE REVIEW

File: src/main/java/Event.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    private String startDate, endDate;
+
+    public Event(String[] args) {
+        super(args[0]);
+        this.taskType = "E";
+        this.startDate = args[1];
+        this.endDate = args[2];
+    }
+    public String printTask(){
+        return this.getTaskIcon() + this.getStatusIcon() + ' ' + this.description + " (from: " + startDate + " to: " + endDate + ")";
```
Comment (BenjaminPoh) [inline src/main/java/Event.java]: This line of code is a little too long (Java coding standard's hard limit is 120 characters). Can consider using line wrapping at appropriate places.

File: src/main/java/TodoList.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    static final String IN_LIST = " in your todo list.\n";
+    static final int MAX_COMMAND_ARGS = 3;
+    static final int MAX_TASKS = 100;
+    static final String BY_COMMAND = " /by";
+    static final int BY_COMMAND_LENGTH = 5;
+    static final String FROM_COMMAND = " /from";
+    static final int FROM_COMMAND_LENGTH = 7;
+    static final String TO_COMMAND = " /to";
+    static final int TO_COMMAND_LENGTH = 5;
+    private static int tasksNumber = 0;
```
Comment (BenjaminPoh) [inline src/main/java/TodoList.java]: This looks like it indicates the number of tasks, but may be misinterpreted to be the index of a specific task. Can consider renaming to numberOfTasks or something similar.

File: src/main/java/TodoList.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    public static void addDeadline(String[] args) {
+        todos[tasksNumber] = new Deadline(args);
+        tasksNumber += 1;
+        System.out.println(LINE_BREAK + HAVE_ADDED + args[0] + TO_LIST + LINE_BREAK);
+    }
+
+    public static void addEvent(String[] args) {
+        todos[tasksNumber] = new Event(args);
+        tasksNumber += 1;
+        System.out.println(LINE_BREAK + HAVE_ADDED + args[0] + TO_LIST + LINE_BREAK);
```
Comment (BenjaminPoh) [inline src/main/java/TodoList.java]: Line 54, Line 60 and Line 66 works similarly. Can consider refactoring to a separate function instead.

File: src/main/java/TodoList.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    public static void mark(int num) {
+        todos[num - 1].isDone = true;
+        System.out.println(LINE_BREAK + MARKED_TASK + num + IN_LIST + todos[num - 1].getStatusIcon()
+                + ' ' + todos[num - 1].description + "\n" + LINE_BREAK);
+    }
+
+    public static void unmark(int num) {
+        todos[num - 1].isDone = false;
+        System.out.println(LINE_BREAK + UNMARKED_TASK + num + IN_LIST + todos[num - 1].getStatusIcon()
+                + ' ' + todos[num - 1].description + "\n" + LINE_BREAK);
```
Comment (BenjaminPoh) [inline src/main/java/TodoList.java]: Line 79-80 is similar in function to Line 85-86. Can consider refactoring to a separate function which updates the status of a specific task instead.

**Review by BenjaminPoh — COMMENTED** [review]
Good effort, great use of Inheritance and refactoring, and code was easy to read/follow. :)

File: src/main/java/shizuka/Deadline.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package shizuka;
+
+public class Deadline extends Todo {
+    private String deadline;
+
+    public Deadline(String[] args) {
+        super(args[0]);
+        this.taskType = "D";
```
Comment (hughjazzman) [inline src/main/java/shizuka/Deadline.java:16]: Perhaps you can avoid the unnecessary use of `this`. Similar for all occurrences.

https://se-education.org/guides/conventions/java/index.html#variables

File: src/main/java/shizuka/Event.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+public class Event extends Todo {
+    private String startDate, endDate;
+
+    public Event(String[] args) {
+        super(args[0]);
+        this.taskType = "E";
+        this.startDate = args[1];
+        this.endDate = args[2];
+    }
+    public String printTask(){
```
Comment (hughjazzman) [inline src/main/java/shizuka/Event.java]:
```suggestion
    public String printTask() {
```

File: src/main/java/shizuka/InvalidCommandFormatException.java [hunk 7]
```diff
@@ -0,0 +1,4 @@
+package shizuka;
+
+public class InvalidCommandFormatException extends Exception{
```
Comment (hughjazzman) [inline src/main/java/shizuka/InvalidCommandFormatException.java:3]:
```suggestion
public class InvalidCommandFormatException extends Exception {
```

File: src/main/java/shizuka/Printer.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
+package shizuka;
+
+public class Printer {
```
Comment (hughjazzman) [inline src/main/java/shizuka/Printer.java]: Nice work separating the Ui code into its own class.

File: src/main/java/shizuka/Shizuka.java [hunk 9]
```diff
@@ -0,0 +1,102 @@
+package shizuka;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Shizuka {
+    static final String FILE_PATH = "shizuka.txt";
+
+    public static String[] parseCommand(String args) {
+        return args.split(" ", 2);
```
Comment (hughjazzman) [inline src/main/java/shizuka/Shizuka.java]: Could this have a magic literal and number?

File: src/main/java/shizuka/Shizuka.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    public static int parseNumber(String args) {
+        int endIndex = args.indexOf(' ') + 1;
+        return Integer.parseInt(args.substring(endIndex));
+    }
+
+    public static void main(String[] args) {
+        Printer.intro();
+        Scanner in = new Scanner(System.in);
+        String line, lineTrimmed;
+        TodoList list0 = new TodoList();
```
Comment (hughjazzman) [inline src/main/java/shizuka/Shizuka.java:12]: Consider using the name to explain, as opposed to adding a number.

https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain

File: src/main/java/shizuka/Shizuka.java [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+        do {
+            line = in.nextLine();
+            lineTrimmed = line.trim();
+            String[] command = parseCommand(lineTrimmed);
+            int taskNum;
+            switch (command[0]) {
```
Comment (hughjazzman) [inline src/main/java/shizuka/Shizuka.java]: Consider having a `Command` class to deal with the command code, to apply more OOP principles and reduce repetitive code.

File: src/main/java/shizuka/TodoList.java [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+    static final String IN_LIST = " in your todo list.\n";
+    static final int MAX_DEADLINE_ARGS = -1;
+    static final int MAX_EVENT_ARGS = 3;
+    static final String BY_COMMAND = " /by ";
+    static final String FROM_COMMAND = " /from ";
+    static final String TO_COMMAND = " /to ";
+    private int numberOfTasks;
+    private final ArrayList<Todo> todos;
+
+    public TodoList(){
```
Comment (hughjazzman) [inline src/main/java/shizuka/TodoList.java]:
```suggestion
    public TodoList() {
```

File: src/main/java/shizuka/TodoList.java [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+        return vars;
+    }
+
+    private static String[] parseEvent(String args) throws InvalidCommandFormatException {
+        String[] vars = new String[MAX_EVENT_ARGS];
+        String[] tempArray0 = args.split(FROM_COMMAND, 2);
+        vars[0] = tempArray0[0];
+        String[] tempArray1 = tempArray0[1].split(TO_COMMAND, 2);
+        vars[1] = tempArray1[0];
+        vars[2] = tempArray1[1];
```
Comment (hughjazzman) [inline src/main/java/shizuka/TodoList.java:53]: As mentioned before, perhaps the names could be more informative here, so the code can be understood directly?

File: src/main/java/shizuka/TodoList.java [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+            Printer.formatError();
+            return;
+        }
+        this.todos.add(new Event(vars));
+        numberOfTasks += 1;
+        Printer.addToList(vars[0]);
+    }
+
+    public void list() {
+        System.out.print(LINE_BREAK + YOUR_TODO_LIST + LINE_BREAK);
```
Comment (hughjazzman) [inline src/main/java/shizuka/TodoList.java:111]: Consider leaving this to the `Printer` class?

**Review by hughjazzman — COMMENTED** [review]
Great effort! Consider adding more classes to apply OOP principles and techniques.

**Pull request closed without merging**